lib/matplotlib/backends/backend_webagg.py

```diff
@@ -54,278 +37,96 @@ class FigureManagerWebAgg(core.FigureManagerWebAgg):
-            self.set_header('Content-Type', 'image/png')
-            self.write(Path(mpl.get_data_path(),
-                            'images/matplotlib.png').read_bytes())
-
-    class SingleFigurePage(tornado.web.RequestHandler):
-        def __init__(self, application, request, *, url_prefix='', **kwargs):
-            self.url_prefix = url_prefix
-            super().__init__(application, request, **kwargs)
-
-        def get(self, fignum):
-            fignum = int(fignum)
-            manager = Gcf.get_fig_manager(fignum)
-
-            ws_uri = f'ws://{self.request.host}{self.url_prefix}/'
-            self.render(
-                "single_figure.html",
-                prefix=self.url_prefix,
-                ws_uri=ws_uri,
-                fig_id=fignum,
-                toolitems=core.NavigationToolbar2WebAgg.toolitems,
-                canvas=manager.canvas)
 
-    class AllFiguresPage(tornado.web.RequestHandler):
-        def __init__(self, application, request, *, url_prefix='', **kwargs):
-            self.url_prefix = url_prefix
-            super().__init__(application, request, **kwargs)
-
-        def get(self):
-            ws_uri = f'ws://{self.request.host}{self.url_prefix}/'
-            self.render(
-                "all_figures.html",
-                prefix=self.url_prefix,
-                ws_uri=ws_uri,
-                figures=sorted(Gcf.figs.items()),
-                toolitems=core.NavigationToolbar2WebAgg.toolitems)
-
-    class MplJs(tornado.web.RequestHandler):
-        def get(self):
-            self.set_header('Content-Type', 'application/javascript')
-
-            js_content = core.FigureManagerWebAgg.get_javascript()
-
-            self.write(js_content)
-
-    class Download(tornado.web.RequestHandler):
-        def get(self, fignum, fmt):
-            fignum = int(fignum)
-            manager = Gcf.get_fig_manager(fignum)
-            self.set_header(
-                'Content-Type', mimetypes.types_map.get(fmt, 'binary'))
-            buff = BytesIO()
-            manager.canvas.figure.savefig(buff, format=fmt)
-            self.write(buff.getvalue())
-
-    class WebSocket(tornado.websocket.WebSocketHandler):
+    class MockPythonWebSocket:
         supports_binary = True
 
+        def __init__(self, manager, js_web_socket):
+            self.manager = manager
+            self.js_web_socket = js_web_socket
+
         def open(self, fignum):
+            self.js_web_socket.open(create_proxy(self.on_message))   # should destroy proxy on close/exit?
             self.fignum = int(fignum)
-            self.manager = Gcf.get_fig_manager(self.fignum)
             self.manager.add_web_socket(self)
-            if hasattr(self, 'set_nodelay'):
-                self.set_nodelay(True)
 
         def on_close(self):
             self.manager.remove_web_socket(self)
 
         def on_message(self, message):
-            message = json.loads(message)
+            message = message.as_py_json()
```

Comment:
Makes sense, assuming that we get a JavaScript JSON object (mix of arrays, objects, numbers, strings, and bools) as input.